tests/README.rst

```diff
@@ -85,3 +89,28 @@ If you want to contribute towards adding a new tests in Fledge, make sure you fo
 
 - Test file name should begin with the word ``test_`` to enable pytest auto test discovery.
 - Make sure you are placing your test file in the correct test directory. For example, if you are writing a unit test, it should be located under ``$FLEDGE_ROOT/tests/unit/python/fledge/<component>`` where component is the name of the component for which you are writing the unit tests. For more information of type of test, refer to the test categories.
+
+Code Coverage
+-------------
+
+Python Tests
+++++++++++++
+
+Fledge uses PyTest-cov Framework of Pytest as the code coverage measuring tool for python tests, For more information on PyTest-cov please refer to |pytest-cov docs|.
+
+To install PyTest-cov Framework along with Pytest Framework use the following command:
+::
+   python3 -m pip install pytest==3.6.4 pytest-cov==2.9.0
+
+Running the python tests:
+
+- ``pytest --cov=. --cov-report xml:xml_filepath --cov-report html:html_directorypath`` - This will execute all the python test files in the given directory and sub-directories and generate the code coverage report in XML as well as the HTML format at the specified path in the command.
+- ``pytest test_filename.py --cov=. --cov-report xml:xml_filepath --cov-report html:html_directorypath`` - This will execute all tests in the file named test_filename.py and generate the code coverage report in XML as well as the HTML format at the specified path in the command.
+- ``pytest test_filename.py::TestClass --cov=. --cov-report xml:xml_filepath --cov-report html:html_directorypath`` -  This will execute all test methods in a single class TestClass in file test_filename.py and generate the code coverage report in XML as well as the HTML format at the specified path in the command.
+- ``pytest test_filename.py::TestClass::test_case --cov=. --cov-report xml:xml_filepath --cov-report html:html_directorypath`` - This will execute test method test_case in class TestClass in file test_filename.py and generate the code coverage report in XML as well as the HTML format at the specified path in the command.
```

Comment:
`--cov-report xml:xml_filepath` - Did it work for you without file? XML path directory should have file `<filename>.xml`

Reply:
If we do not provide the XML path, it will generate a `coverage.xml` file in the location from which the test is executed.

Reply:
Python 3.6.9 + pytest (3.6.4) + pytest-cov(2.9.0)

`IsADirectoryError: [Errno 21] Is a directory:`

Reply:
Yes, it is working fine with Python 3.6.9 + pytest (3.6.4) + pytest-cov(2.9.0)

Logs:
```
ubuntu@ip-10-0-0-30:~/fledge$ python3 -m pytest -s -vv --junit-xml=$FLEDGE_ROOT/test_output_unit.xml $FLEDGE_ROOT/tests/unit/python/fledge/common/test_alert_manager.py::TestAlertManager::test_urgency --tb=line --cov=. --cov-report xml --cov-report html:${HOME}/python_test_coverage/unit_tests
================================================ test session starts =================================================
platform linux -- Python 3.6.9, pytest-3.6.4, py-1.11.0, pluggy-0.7.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/ubuntu/fledge, inifile:
plugins: cov-2.9.0, mock-1.10.3, allure-adaptor-1.7.10, asyncio-0.10.0, aiohttp-0.3.0
collected 1 item

tests/unit/python/fledge/common/test_alert_manager.py::TestAlertManager::test_urgency PASSED

---------------------------- generated xml file: /home/ubuntu/fledge/test_output_unit.xml ----------------------------

----------- coverage: platform linux, python 3.6.9-final-0 -----------
Coverage HTML written to dir /home/ubuntu/python_test_coverage/unit_tests
Coverage XML written to file coverage.xml

============================================== 1 passed in 5.63 seconds ==============================================
ubuntu@ip-10-0-0-30:~/fledge$
ubuntu@ip-10-0-0-30:~/fledge$
ubuntu@ip-10-0-0-30:~/fledge$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.6 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
ubuntu@ip-10-0-0-30:~/fledge$ python3 -V; python3 -m pip list | grep pytest
Python 3.6.9
pytest                3.6.4
pytest-aiohttp        0.3.0
pytest-allure-adaptor 1.7.10
pytest-asyncio        0.10.0
pytest-cov            2.9.0
pytest-mock           1.10.3
```

Reply:
Then it may be enviromental problem